sigar_linux_common_test.go

```diff
@@ -0,0 +1,101 @@
+package gosigar_test
+
+import (
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	sigar "github.com/elastic/gosigar"
+	"github.com/stretchr/testify/assert"
+)
+
+var mtabDir string
+var oldMtabf string
+
+func setUpCommonTest(t testing.TB) {
+	var err error
+	oldMtabf = sigar.Mtabf
+	mtabDir, err = ioutil.TempDir("", "sigarTests")
+	sigar.Mtabf = mtabDir + "/mtab"
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
You can use `t.TempDir`, that creates a temporary directory that is removed after the test is executed.
And it would be safer to use `filepath.Join()` to build the full path.
```suggestion
	mtabDir, err = t.TempDir()
	sigar.Mtabf = filepath.Join(mtabDir, mtab)
```